### DISCUSSION TITLE
Wie sieht die Zukunft mit VCA und LS25 aus? // What does the future hold for VCA and LS25?

### DISCUSSION THREAD

**Discussion by bpNetwork** [Mogli12/KeyboardSteer#921] [General]
Ich würde viele Funktionen vom VCA im LS25 vermissen. Wie z.B. die Blickrichtung nach dem Lenkrad oder der Spurwechsel per Knopfdruck in die nächste Bahn, dies kann ja das eingebaute LS25 GPS nicht.
Könnt ihr schon etwas zu der Zukunft vom VCA im LS25 sagen?
//
I would miss many functions of the VCA in the LS25. For example, the direction of view after the steering wheel or the lane change to the next lane at the touch of a button, which the built-in LS25 GPS cannot do.
Can you say anything about the future of the VCA in the LS25?

Björn

**Comment by ElfVierzehn** [comment]
I am interested, too. Steering is quite impossible with mouse and keyboard, and in its current state steering with controller is even worse. Steering, keep speed, as well as separate buttons for rear and front tools lowering and lifting are my absolutely favorite functions and I have to give in, I really miss the mod in general.

**Comment by SbSh-DNA-ModDing** [comment]
Lasst dem Mogli doch erstmal Luft holen Leute.
Der realease war am besten schon vor 2 Wochen oder wie?
Ich bin mir sicher er wird kommen, nur Geduld!


First let Mogli breathe, people.
The real release was best already 2 weeks ago or what?
I'm sure it will come, just be patient!

**Comment by ElfVierzehn** [comment]
That's not meant to perform any pressure. It's meant to show interest, that could help if he is e. g. still deciding if he actually wants to bring it over to FS25 or not.

**Comment by Sirtrucker** [comment]
War/ist eine toller mod. Vermisse ihn im 25ziger auch. Würde mich freuen, wenn dieses Projekt "am Leben bleibt".

**Comment by Mogli12** [comment]
Hi all,

In the last month I started playing iRacing. Thus, my focus shifted a little. 
The good news are that I bought the new FS25. I think it has a lot of potential, but some things a clearly missing. 
The first things I plan to look into are the modifications of the camera and cruise control. I hope I could do it without script docu. 
And the clutch in FS25 needs a lot of improvements. But that requires the script docu ob GDN. 

That's all for now,
Stefan

**Reply by Tubezki** [reply]
Free Steering mode is a must for us controller players, its literally the mod that makes the game playable for me, without VCA, I cant stand the default steering for controllers.

**Reply by dasich86** [reply]
The VCA mod is a must-have for me. The camera module alone is a blessing. I play with the steering wheel and side console. In the interior view. Looking around with the mouse is really not nice. VCA = Great.

**Reply by Robertszabo1991** [reply]
hello, I'm also desperately waiting for the free steering function, have you found out anything in all this time, the free steering function has appeared?, thank you

**Comment by Mogli12** [comment]
Hi all,

Some things still work as before. But the settings UI is not working at all. The is always a lot of re-engineering and I do not want to go this way again. 
I think that I will split VCA into several mods. One mod handling the camera rotations. A second mod around GPS. Then the transmission needs a complete overhaul. I have now a shifter with 7+R gears and these strange 1L, 1H, ..., 4H transmissions do not work for me. It is just no fun to map the reverse gear of my shifter (upper left) to the eight gear. And it feels unrealistic because the gear stick ingame acts differently.

BR, Stefan

**Reply by SbSh-DNA-ModDing** [reply]
These strange powershift transmissions have no clutch! and the groupTypes, gearTypes numbers have changed slightly.
It could be the thing that it don't work scriptside cuz of this.
The Modul like mod, wouldn't be so stupid - I think it's a good idea.

With the UI.. DBL will do it in the future, to show the diff awd etc.

**Reply by Helgomant** [reply]
Hallo Stefan, erstmal vielen Dank, dass du dich noch mal dem Getriebe im LS25 widmen willst. lt. Giants Entwickler wurde an der Getriebe-/Kupplungsfunktion zwischen dem LS 22 und 25 nichts geändert. Leider! Heißt aber im Umkehrschluss, dass hoffentlich nicht allzu viel Arbeit bei der Konvertierung entsteht. Wie du schon erwähnt hast, funktioniert die Kupplung im 25er nicht richtig. Fahrzeuge können nicht im Standgas gefahren werden und der Motor geht aus. Was deutlich verbessert wurde, ist das Ausrollverhalten. Das kommt der Realität schon deutlich näher, als noch im 22er.
Bei den manuellen Getrieben fehlt vor allen Dingen eine dritte Konfigurationsmöglichkeit für weitere Gänge. 
Als Beispiel der John Deere 3650.
Gruppe für R, I, II
Gänge 1 bis 4
Lastschaltstufen High und Low
Da Giants, genau wie im 22er, keine Möglichkeit für eine separate Konfiguration der Lastschaltstufen eingebaut hat, haben sie diese einfach mit in die Gänge gepackt (war im 22er beim Valtra 8150 genauso). 1L, 1H, 2L usw. Im realen Trecker kannst du die LS-Stufen und die Gänge unabhängig voneinander schalten.
Ich habe in meinem Setup zwei H-Shifter. Das bedeutet, ich kann das original Getriebe bis auf die LS-Stufen abbilden. Die Low Stufen der Gänge habe ich in der XML aus kommentiert. Fahre also nur 1H, 2H, 3H und 4H. Die Übersetzungen passen leider auch nicht mit den Originalen überein. Zumindest in Europa laufen die 3650 gute 40 km/h.

Die Aufsplittung von VCA halte ich übrigens für eine gute Idee. Die Kamera und GPS Funktion habe ich zum Beispiel in der Vergangenheit nie genutzt (benutze Headtracking und mir reicht vermutlich sogar die GPS Funktion im LS25). 

Viele Grüße
Helge

**Reply by SbSh-DNA-ModDing** [reply]
Ja Helge, was die sagen und das was ist, sind vielleicht die selbsen Schuhe, aber unterschiedliche Schnürsenkel.
Die Types sind anders angeordnet und bei den "nur" Lastgeschalteten gibt es keine Kupplung mehr als Funktion. (clutch / clutchPedal)
Sam's realismAddon_Gearbox ist meine ich auch in der Mache... ob und wann das released wird weiß ich aber nicht.
Dennoch die Diff und Allrad Funktionen, Geräte hinten / vorne getrennt ansteuern.. sowas würde mir fehlen.

btw. cvt-addon wird auch noch etwas dauern.. es hat sich doch einiges geändert, ich komme ohne doku nicht weiter.
Lg Sebastian

**Reply by Helgomant** [reply]
Hi Sebastian, stimmt das mit der Kupplung ist mir heute beim Claas Arion auch aufgefallen. Das Getriebe (Gruppen und Lastschaltstufen) ist ansonsten leider auch völlig vermurkst. Ich frage mich, warum im Code immer alles geändert wird, wenn sich ansonsten nicht viel getan hat?! Wenn ich im JD 3650 die Kupplung beim Rückwärtsfahren trete bleibt der Trecker sofort stehen und rollt nicht aus, so wie im Vorwärtsgang. Das muss man auch nicht verstehen.
Das Sam bereits an seinem Gearbox Addon rumprobiert, hatte ich bereits in einem seiner Videos mitbekommen. Ohne Script Doku wird es vermutlich bei ihm aber auch nicht weiter gehen. Hoffen wir mal, das Giants damit schnell um die Ecke kommt.
VG
Helge

**Reply by SbSh-DNA-ModDing** [reply]
Ja, die Doku brauchen wir so oder so.

**Reply by k4talin** [reply]
Small question.. Why this mod need to interfere with gearboxes? IMO i just need a mod to control steering. I would not expect VCA to do the Course play job at any point ,,, Also we have AD for that. I do not get it. Appologise for question . ...I am not a moding guy . Especially after i read gearboxes gives you most trouble. If you can release something to keep tractors on a specific chosen line i guess everyone will be happy. Good Luck and respect.

**Reply by iljarogoff** [reply]
Hi, i really hope for a GPS help, driving straight lines outside fields is something I really miss, e.g. for making AD-courses. Thx for all your work you have done and everything that will come in future!

**Comment by Harag9** [comment]
Just would like to say that I miss this mod so much in FS25, loved it in 22, and 19. Steering with k/m is so much smoother with this. Can't wait for a new version, but I know it takes time, and I appreciate the work you do. I'll keep an eye out for it on the game Modhub :)

**Reply by k4talin** [reply]
> Just would like to say that I miss this mod so much in FS25, loved it in 22, and 19. Steering with k/m is so much smoother with this. Can't wait for a new version, but I know it takes time, and I appreciate the work you do. I'll keep an eye out for it on the game Modhub :)

For now i would not care about km/hr. Just steering is fine. There is nothing to help us drive on a lane WE CHOOSE

**Comment by Hinki93** [comment]
Hi Mogli, 
Vielen Dank vorab für deine Mühen mit dem VCA im LS19 und LS22. 
Da du ja überlegst die Mod zu splitten.... ich brauche eigentlich nur die Funktion das vordere und hintere Gerät getrennt per Knopfdruck zu steuern, es würde sogar reichen diese nur getrennt senken/heben zu können. Den Rest nutze ich garnicht.
Ich versteh nicht warum Giants das nicht von Haus aus anbietet... Aber so Dinge wie Haspel rauf/runter und ein/ausfahren welche keinen praktischen Nutzen haben, machen sie schon rein.
Falls das nicht viel Arbeit ist, diese eine Funktion als Mod zu bringen, würd ich dafür sogar eine Spende leisten ;)
Schöne Grüße
Hinki

**Reply by Haurshi** [reply]
Also bei mir kann man das vorder und hintere Gerät heben und senken wie man will .

**Reply by Hinki93** [reply]
Musst du das entsprechende Gerät dafür in der Geräteauswahl oben links extra auswählen?

Edit: ich will die Funktion mit Taste 17 vorne senken/heben, Taste 18 vorne an/aus Taste 19 hinten senken/heben, Taste 20 hinten an/aus

**Reply by Haurshi** [reply]
achso ja ne das is natürlich bisl was anderes 
Nur aus Intresse was machst du das du es so brauchst ?

**Reply by Hinki93** [reply]
Zum Beispiel beim Mähen die Mähwerke heben, da mir das automatische Zeitversetzte heben zum einen zu langsam geht und zum anderen manchmal erst das hintere und dann das vordere gehoben wird. Oder auch vorne mähen, hinten Ladewagen.
Im realen Leben hebt man das Pickup des Ladewagens auch am Ende der Spur.

Im LS19 wurde beim durchschalten der Geräte nicht auf den Traktor gewechselt, da war das noch annehmbar weil man einfach nur einmal auf wechseln und dann eben auf senken/heben gedrückt hat.

**Reply by Haurshi** [reply]
sry aber das is doch ein Luxusproblem , nimm odch alles heben alles senken, das geht doch super

**Reply by Hinki93** [reply]
Wenn das für dich als Luxus zählt... Deine Sache... Ich spiele mit Lenkrad, zwei Joysticks, Seitenkonsole und Headtracking, um möglichst nah an die Realität zu kommen, da ich leider in der Realität nicht mehr die Möglichkeit habe die Landwirtschaft auszuüben. Deshalb möchte ich auch die Funktionen wie in der Realität haben. Und wie ich oben schon geschrieben habe, Haspel beim Mähdrescher steuern geht im Spiel hat aber keinen Nutzen, aber vorderes und hinteres Gerät mit eigener Belegung zu steuern geht nicht ab Werk, was durchaus einen Nutzen bei entsprechender Spielweise hat. Völlig unverständlich für mich.
Aber wenn du mit der Automatik zufrieden bist, freut mich das natürlich für dich :)

**Comment by k4talin** [comment]
@Mogli12 Any news Sir? Pls. I get depression with ingame AI

**Reply by Haurshi** [reply]
13.01.2025 and no news  ?

**Comment by cabmac** [comment]
Moin...also ich habe alles in allem vielleicht 2 Stunden den 25er ausprobiert, da gehen mir soviele Dinge auf den Senkel aber was ich persoenlich als Tastaturspieler am meisten vermisse ist deine "vehicleControlTastatur"-geschwindigkeitsabhaengige Steuerung und das Inching.....Das ist wirklich ein grauss, ohne damit zu spielen, weshalb ich erstmal schoen beim LS22 bleibe... ;)
Schoene Gruesse,

**Reply by Haurshi** [reply]
kauf dir halt ma en Xbox Controller , so viel kosten die auch nich und es macht viel mehr Spaß

**Reply by cabmac** [reply]
So ein BS....erstens habe ich einen und zweitens kann ich das mit dem LEnkrad ja noch einigermassen verstehen, aber einen Controller? Sich mit dem Stickl umsehen...lass mal stecken..ausserdem wuesste ich nicht, was die Diskussion hier zu suchen hat???

**Comment by Haurshi** [comment]
Naja da sich der Modder nicht mehr meldet , kann sich ja ein anderer der Sache annehmen oder ?
Wäre schade drum wenn es kein VC mehr gäbe.

**Comment by sootysax** [comment]
The sounds in GPS when activating the line and when you come to the end of a lane is sorely missing!

**Comment by BetaTGC** [comment]
man i miss the free steering smoothness and the keep speed, its extremely hard to play fs without vca, cant wait for it to be brought to fs25. thank you Mr Mogli for this amazing mod :)

**Comment by Tubezki** [comment]
Some kind of update would be really nice, at least to know that is there any active developing going or is this project end of life? Please, could you update at least that if its inactive or "eol".

**Comment by iljarogoff** [comment]
Yeehaw! Realized that there is a alpha VCA for FS25! I love it so much!!! you are the best. Thanks a lot!

**Comment by cabmac** [comment]
very nice...so let`s reinstall FS25.. ;D

**Comment by Williwillswisse** [comment]
Now life has meaning again 😉

**Reply by Funkyklaus** [reply]
Hallo,
freue mich wenn es komplett ist

**Comment by Robertszabo1991** [comment]
hello, I also found the vca alpha version but I can't find the free steering function, can someone help me activate it, thanks

**Comment by Falkeger** [comment]
Hallo Zusammen,

kann mir einer sagen was das verursacht ?
Script error: Divide by zero error.
LUA call stack:
  =D:/home/sid_8504897/farmingSim_2025/profile/mods/FS25_VehicleControlAddon/lua/vehicleControlAddon.lua:3382 onPostUpdate
  =dataS/scripts/specialization/SpecializationUtil.lua:56 raiseEvent
  =dataS/scripts/vehicles/Vehicle.lua:2045 update
  =dataS/scripts/network/NetworkNode.lua:181 updateActiveObjects
  =dataS/scripts/network/Server.lua:64 update
  =dataS/scripts/BaseMission.lua:1023 update
  =dataS/scripts/utils/Utils.lua:604 
  =dataS/scripts/FSBaseMission.lua:1913 update
  =dataS/scripts/mission00.lua:683 update
  =dataS/scripts/main.lua:1240 update
2025-12-28 19:54:13.046
